qemu/tests/qemu_guest_agent.py

```diff
@@ -1413,6 +1413,36 @@ def _value_compared_ga_guest(return_value_ga,
 
         _prepared_n_restore_env(False)
 
+    @error_context.context_aware
+    def gagent_check_get_cpustats(self, test, params, env):
+        """
+        Execute "guest-get-cpustats" command to guest agent.
+
+        Steps:
+        1) Check cpu stats info
+        2) Check number of cpus.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        session = self._get_session(params, self.vm)
+        self._open_session_list.append(session)
+
+        error_context.context("Check cpustats info of guest and "
+                              "number of cpus.", LOG_JOB.info)
+        cs_info_qga = self.gagent.get_cpustats()
+        cpu_num_guest = int(session.cmd_output(params["cpu_num_guest"]))
+        cpu_num_qga = 0
+        for cs in cs_info_qga:
+            cpu_num_qga += 1
+            for key in cs.keys():
+                if (key != "type" and key != "cpu" and
+                        key not in params["cpustats_info_list"]):
+                    test.fail("Cpustats info is not totally correct.")
```

Comment:
Can we record all in proper fields and fail them together after finish the whole `for` loop for debug purpose?
And I think we need to make sure `type` is correct per guest OS, even now it only support 'linux'.
I saw it is 'only Linux', do we know what happened if execute it on Windows guest?

Reply:
> 
1. Q: Can we record all in proper fields and fail them together after finish the whole for loop for debug purpose? 
A: you mean don't compare them one by one, instead, compare the whole fields one time? 
2. Q: I think we need to make sure type is correct per guest OS 
A: okay, sure. I'll add this part later.
3. Q: do we know what happened if execute it on Windows guest?
A: it will give us: {"error": {"class": "GenericError", "desc": "this feature or command is not currently supported"}}

Reply:
I mean you will not go through all strings once one of them are failing the case.
```suggestion
        improper_list = []
        for cs in cs_info_qga:
            cpu_num_qga += 1
            for key in cs.keys():
                if (key != "type" and key != "cpu" and
                        key not in params["cpustats_info_list"]):
                    improper_list.append({cs["cpu"]: key})
        if fail_list:
            test.fail("Cpustats info has improper elements: %s" % improper_list)
```

Reply:
okay， got it.

Reply:
(1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_get_cpustats.q35: PASS (85.42 s)